src/scripts/core/models/column.ts

```diff
@@ -37,6 +37,17 @@ export class Column extends Container{
         return this.tokens;
     }
 
+    public removeAllTokens(): void {
+        this.tokens = [];
+    }
+
+    public replaceAllTokens(newTokens: Token[]): void {
+        if(newTokens.length != this.tokens.length) {
+            console.log("error on replaceAllTokens: input length does not match");
+            return;
```

Comment:
Looks like this is reporting an error. console.log is fine for basic reporting while developing but there are a couple alternatives you may not yet know about.

- `console.error()` - basically the same as log, but it shows as red in the console to highlight the issue better.
- `throw new Error('message')` or just `throw 'message'` - the technically correct way to handle errors as it allows you to detect and handle your errors with a `try {} catch {}` block like you've probably seen in other languages.